corrector.rb

```diff
@@ -0,0 +1,28 @@
+# Class Corrector
+# Developed by: Alex Castillo
+#
+# Description: decorator class to decorate person class, and fix the name to starts with capital letter,
+# and not longer than 10 characters
+#
+# Example:
+#
+# person_object1 = Person.new(name: "john", age: 20, parent_permission: true)
+# corrector_object1 = Corrector.new(person_object1)
+# person_object1.name # => "John"
+#
+# Date: Dec, 22 2021
+
+require './person'
+
+class Corrector
+  def initialize(person: Person)
+    person.name = correct_name(person.name)
```

Comment:
- [ ] There should be an instance of Corrector set in the Person class on initialization
- [ ] We should have `validate_name` method in the `Person` class.

Reply:
Ohh, you are right, I didn't look in-depth to that requirements, anyway, those requirements, are not usual to a decorator pattern, and make you violate the Solid principles.